### PR TITLE
Fix fatal in filter_upgrader_source_selection()

### PR DESCRIPTION
**Pull request by defunctl**
- Cherry picked from https://github.com/stellarwp/uplink/pull/65
- Fixes tests by using a theme WordPress still ilncludes.

### CODE REVIEW

**Review by tarecord — APPROVED** [review]
Looks good to me! Nothing that would keep this from being released, but I would like to see a test for this 😄

File: src/Uplink/Admin/Update_Prevention.php [hunk 1]
```diff
@@ -50,14 +50,14 @@ public function is_stellar_uplink_resource( string $plugin ): bool {
 	 *
 	 * @since  4.9.12
 	 *
-	 * @param string       $source        File source location.
-	 * @param mixed        $remote_source Remote file source location.
-	 * @param WP_Upgrader  $upgrader      WP_Upgrader instance.
-	 * @param array<mixed> $extras         Extra arguments passed to hooked filters.
+	 * @param string|WP_Error $source        File source location.
```
[the remark this answers is not in the document]
Comment (defunctl) [reply src/Uplink/Admin/Update_Prevention.php:53]: Yeah, this entire feature needs better test coverage but we'll have to introduce that later. Ben's been using this in the wild for a bit now.

File: src/Uplink/Admin/Provider.php [hunk 2]
```diff
@@ -150,11 +150,11 @@ public function store_admin_notices( $page ): void {
...
-	 * @param bool         $reply      Whether to bail without returning the package.
-	 *                                 Default false.
-	 * @param string       $package    The package file name or URL.
-	 * @param \WP_Upgrader $upgrader   The WP_Upgrader instance.
-	 * @param array        $hook_extra Extra arguments passed to hooked filters.
+	 * @param  bool|\WP_Error  $reply       Whether to bail without returning the package.
+	 *                                      Default false.
+	 * @param  string          $package     The package file name or URL.
+	 * @param  \WP_Upgrader    $upgrader    The WP_Upgrader instance.
+	 * @param  array           $hook_extra  Extra arguments passed to hooked filters.
```
Comment (tarecord) [inline src/Uplink/Admin/Provider.php:157]: Two spaces between `@param` and the type?

Comment (defunctl) [reply src/Uplink/Admin/Provider.php:157]: It's equal alignment docblock alignment, like the method below: https://github.com/stellarwp/uplink/blob/ec4b0970ed95202c923dc9df2e1abc2b95b9c252/src/Uplink/Admin/Provider.php#L179-L191

File: src/Uplink/Admin/Update_Prevention.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (tarecord) [reply src/Uplink/Admin/Update_Prevention.php:53]: I figured that was the case

**Pull request merged**